File: reference/SmartPharmacy-Kotlin/app/src/main/java/com/example/ui/screens/ProfileScreen.kt
package com.example.ui.screens

import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material.icons.outlined.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.components.ConfirmationDialog
import com.example.ui.components.CustomButton
import com.example.ui.theme.*
import com.example.ui.viewmodel.PharmacyViewModel

@Composable
fun ProfileScreen(
    viewModel: PharmacyViewModel,
    onLogout: () -> Unit
) {
    val context = LocalContext.current
    val email by viewModel.userEmail.collectAsState()

    var showLogoutDialog by remember { mutableStateOf(false) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(SurfaceWhite)
            .padding(16.dp)
            .verticalScroll(rememberScrollState()),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Shop Profile Header
        Card(
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(containerColor = PrimaryBlueLight),
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(20.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Box(
                    modifier = Modifier
                        .size(80.dp)
                        .clip(CircleShape)
                        .background(PrimaryBlue),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.LocalPharmacy,
                        contentDescription = "Shop Logo",
                        tint = SurfaceWhite,
                        modifier = Modifier.size(44.dp)
                    )
                }

                Spacer(modifier = Modifier.height(12.dp))

                Text(
                    text = "Smart Care Pharmacy",
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold,
                    color = TextPrimary
                )

                Text(
                    text = "License No: 20B/MH-102938",
                    fontSize = 12.sp,
                    color = TextSecondary
                )

                Spacer(modifier = Modifier.height(8.dp))

                Surface(
                    shape = RoundedCornerShape(20.dp),
                    color = PrimaryBlue
                ) {
                    Text(
                        text = "Owner: Dr. Alex Morgan",
                        fontSize = 12.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = SurfaceWhite,
                        modifier = Modifier.padding(horizontal = 12.dp, vertical = 4.dp)
                    )
                }
            }
        }

        // Contact & Shop Info
        Card(
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(containerColor = SurfaceWhite),
            elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
            modifier = Modifier
                .fillMaxWidth()
                .border(1.dp, SurfaceBorder, RoundedCornerShape(16.dp))
        ) {
            Column(
                modifier = Modifier.padding(16.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                Text("Pharmacy Credentials & Contact", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = TextPrimary)

                HorizontalDivider(color = SurfaceBorder)

                ProfileInfoRow(icon = Icons.Outlined.Email, label = "Pharmacist Email", value = email)
                ProfileInfoRow(icon = Icons.Outlined.Phone, label = "Store Contact", value = "+91 98200 44556")
                ProfileInfoRow(icon = Icons.Outlined.LocationOn, label = "Store Address", value = "Plot 12, Healthcare Zone, City Center")
                ProfileInfoRow(icon = Icons.Outlined.Receipt, label = "GSTIN", value = "27ABCDE1234F1Z5")
            }
        }

        // Settings Actions List
        Card(
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(containerColor = SurfaceWhite),
            elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
            modifier = Modifier
                .fillMaxWidth()
                .border(1.dp, SurfaceBorder, RoundedCornerShape(16.dp))
        ) {
            Column(modifier = Modifier.padding(8.dp)) {
                SettingsOptionItem(
                    icon = Icons.Outlined.Settings,
                    title = "System Settings",
                    subtitle = "Low stock alert threshold, GST configuration",
                    onClick = { Toast.makeText(context, "System Settings Opened", Toast.LENGTH_SHORT).show() }
                )

                HorizontalDivider(color = SurfaceBorder)

                SettingsOptionItem(
                    icon = Icons.Outlined.Backup,
                    title = "Data Backup & Export",
                    subtitle = "Export local dummy database to CSV",
                    onClick = { Toast.makeText(context, "Exporting pharmacy database to CSV...", Toast.LENGTH_SHORT).show() }
                )

                HorizontalDivider(color = SurfaceBorder)

                SettingsOptionItem(
                    icon = Icons.Outlined.HelpOutline,
                    title = "Help & Support",
                    subtitle = "Documentation and user guide",
                    onClick = { Toast.makeText(context, "Smart Pharmacy Help Guide Loaded", Toast.LENGTH_SHORT).show() }
                )
            }
        }

        Spacer(modifier = Modifier.height(8.dp))

        // Logout Button
        CustomButton(
            text = "Logout",
            containerColor = AlertRedLight,
            contentColor = AlertRed,
            icon = Icons.Default.Logout,
            onClick = { showLogoutDialog = true }
        )
    }

    if (showLogoutDialog) {
        ConfirmationDialog(
            title = "Logout from System?",
            message = "Are you sure you want to end your session? You will return to the Login screen.",
            confirmText = "Logout",
            onConfirm = {
                viewModel.logout()
                onLogout()
            },
            onDismiss = { showLogoutDialog = false }
        )
    }
}

@Composable
private fun ProfileInfoRow(
    icon: ImageVector,
    label: String,
    value: String
) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier.fillMaxWidth()
    ) {
        Icon(imageVector = icon, contentDescription = null, tint = PrimaryBlue, modifier = Modifier.size(20.dp))
        Spacer(modifier = Modifier.width(12.dp))
        Column {
            Text(text = label, fontSize = 11.sp, color = TextSecondary)
            Text(text = value, fontSize = 13.sp, fontWeight = FontWeight.SemiBold, color = TextPrimary)
        }
    }
}

@Composable
private fun SettingsOptionItem(
    icon: ImageVector,
    title: String,
    subtitle: String,
    onClick: () -> Unit
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(10.dp))
            .clickable { onClick() }
            .padding(12.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Box(
            modifier = Modifier
                .size(38.dp)
                .clip(CircleShape)
                .background(PrimaryBlueLight),
            contentAlignment = Alignment.Center
        ) {
            Icon(imageVector = icon, contentDescription = null, tint = PrimaryBlue, modifier = Modifier.size(20.dp))
        }

        Spacer(modifier = Modifier.width(12.dp))

        Column(modifier = Modifier.weight(1f)) {
            Text(text = title, fontSize = 14.sp, fontWeight = FontWeight.Bold, color = TextPrimary)
            Text(text = subtitle, fontSize = 12.sp, color = TextSecondary)
        }

        Icon(imageVector = Icons.Default.ChevronRight, contentDescription = null, tint = TextMuted)
    }
}

File: lib/screens/profile_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../core/theme/app_colors.dart';
import '../providers/pharmacy_provider.dart';
import '../routes/app_routes.dart';
import '../widgets/common/custom_button.dart';
import '../widgets/common/confirmation_dialog.dart';

class ProfileScreen extends StatelessWidget {
  const ProfileScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<PharmacyProvider>();

    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(16.0),
          child: Column(
            children: [
              // Shop Profile Header
              _buildProfileHeader(),
              const SizedBox(height: 16),

              // Contact & Shop Info Card
              _buildCredentialsCard(provider.userEmail),
              const SizedBox(height: 16),

              // Settings Actions List
              _buildSettingsCard(context),
              const SizedBox(height: 16),

              // Logout Button
              CustomButton(
                text: "Logout",
                containerColor: AppColors.alertRedLight,
                contentColor: AppColors.alertRed,
                icon: Icons.logout,
                onPressed: () => _showLogoutDialog(context, provider),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildProfileHeader() {
    return Card(
      elevation: 0,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
      color: AppColors.primaryBlueLight,
      child: Container(
        width: double.infinity,
        padding: const EdgeInsets.all(20.0),
        child: Column(
          children: [
            Container(
              width: 80,
              height: 80,
              decoration: const BoxDecoration(
                color: AppColors.primaryBlue,
                shape: BoxShape.circle,
              ),
              child: const Icon(
                Icons.local_pharmacy,
                color: Colors.white,
                size: 44,
              ),
            ),
            const SizedBox(height: 12),
            const Text(
              "Smart Care Pharmacy",
              style: TextStyle(
                fontSize: 20,
                fontWeight: FontWeight.bold,
                color: AppColors.textPrimary,
              ),
            ),
            const Text(
              "License No: 20B/MH-102938",
              style: TextStyle(fontSize: 12, color: AppColors.textSecondary),
            ),
            const SizedBox(height: 8),
            Container(
              padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 4),
              decoration: BoxDecoration(
                color: AppColors.primaryBlue,
                borderRadius: BorderRadius.circular(20),
              ),
              child: const Text(
                "Owner: Dr. Alex Morgan",
                style: TextStyle(
                  fontSize: 12,
                  fontWeight: FontWeight.w600,
                  color: Colors.white,
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildCredentialsCard(String email) {
    return Card(
      elevation: 1,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
        side: const BorderSide(color: AppColors.surfaceBorder),
      ),
      color: AppColors.surfaceWhite,
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const Text(
              "Pharmacy Credentials & Contact",
              style: TextStyle(fontWeight: FontWeight.bold, fontSize: 14),
            ),
            const Divider(height: 24, color: AppColors.surfaceBorder),
            _buildInfoRow(Icons.email_outlined, "Pharmacist Email", email),
            const SizedBox(height: 12),
            _buildInfoRow(Icons.phone_outlined, "Store Contact", "+91 98200 44556"),
            const SizedBox(height: 12),
            _buildInfoRow(Icons.location_on_outlined, "Store Address", "Plot 12, Healthcare Zone, City Center"),
            const SizedBox(height: 12),
            _buildInfoRow(Icons.receipt_outlined, "GSTIN", "27ABCDE1234F1Z5"),
          ],
        ),
      ),
    );
  }

  Widget _buildInfoRow(IconData icon, String label, String value) {
    return Row(
      children: [
        Icon(icon, color: AppColors.primaryBlue, size: 20),
        const SizedBox(width: 12),
        Expanded(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text(label, style: const TextStyle(fontSize: 11, color: AppColors.textSecondary)),
              Text(
                value,
                style: const TextStyle(fontSize: 13, fontWeight: FontWeight.w600),
                maxLines: 1,
                overflow: TextOverflow.ellipsis,
              ),
            ],
          ),
        ),
      ],
    );
  }

  Widget _buildSettingsCard(BuildContext context) {
    return Card(
      elevation: 1,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
        side: const BorderSide(color: AppColors.surfaceBorder),
      ),
      color: AppColors.surfaceWhite,
      child: Padding(
        padding: const EdgeInsets.all(8.0),
        child: Column(
          children: [
            _buildSettingsItem(
              context,
              Icons.settings_outlined,
              "System Settings",
              "Low stock alert threshold, GST configuration",
            ),
            const Divider(height: 1, color: AppColors.surfaceBorder),
            _buildSettingsItem(
              context,
              Icons.backup_outlined,
              "Data Backup & Export",
              "Export local dummy database to CSV",
            ),
            const Divider(height: 1, color: AppColors.surfaceBorder),
            _buildSettingsItem(
              context,
              Icons.help_outline,
              "Help & Support",
              "Documentation and user guide",
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildSettingsItem(BuildContext context, IconData icon, String title, String subtitle) {
    return InkWell(
      onTap: () {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text("$title feature coming soon")),
        );
      },
      borderRadius: BorderRadius.circular(10),
      child: Padding(
        padding: const EdgeInsets.all(12.0),
        child: Row(
          children: [
            Container(
              width: 38,
              height: 38,
              decoration: const BoxDecoration(
                color: AppColors.primaryBlueLight,
                shape: BoxShape.circle,
              ),
              child: Icon(icon, color: AppColors.primaryBlue, size: 20),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(title, style: const TextStyle(fontSize: 14, fontWeight: FontWeight.bold)),
                  Text(subtitle, style: const TextStyle(fontSize: 12, color: AppColors.textSecondary)),
                ],
              ),
            ),
            const Icon(Icons.chevron_right, color: AppColors.textMuted),
          ],
        ),
      ),
    );
  }

  void _showLogoutDialog(BuildContext context, PharmacyProvider provider) {
    showDialog(
      context: context,
      builder: (ctx) => ConfirmationDialog(
        title: "Logout from System?",
        message: "Are you sure you want to end your session? You will return to the Login screen.",
        confirmText: "Logout",
        isDanger: true,
        onConfirm: () {
          provider.logout();
          Navigator.pushNamedAndRemoveUntil(context, AppRoutes.login, (route) => false);
        },
        onDismiss: () {},
      ),
    );
  }
}
